steps/aws-secrets-manager/1.0.0/step.yml

```diff
@@ -0,0 +1,87 @@
+title: Fetch secrets from AWS Secrets Manager
+summary: Fetch secrets from AWS Secrets Manager
+description: |
+  The main intention of this Step is to allow storage of sensitive data on AWS Secrets Manager, and fetching them on-demand during Bitrise workflow execution.
+
+  When a workflow runs, the Step fetches a list of specified secrets from AWS Secrets Manager.
+
+  The fetched secrets are then propagated into subsequent steps in the workflow.
+
+  The Step uses AWS SDK for Go v2 with the default config loader. This means for authenticating with AWS, you may:
+
+    - Use static AWS credentials via environment variable, e.g. `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`.
+    - Use shared configuration files, e.g. `AWS_PROFILE`.
+
+  Checkout the main repository README for more usage guide.
+website: https://github.com/MoneyLion/bitrise-step-aws-secrets-manager
+source_code_url: https://github.com/MoneyLion/bitrise-step-aws-secrets-manager
+support_url: https://github.com/MoneyLion/bitrise-step-aws-secrets-manager/issues
+published_at: 2021-07-21T21:06:31.545432+08:00
+source:
+  git: https://github.com/MoneyLion/bitrise-step-aws-secrets-manager.git
+  commit: 18db9fda4e3bbeb88b47e75f4bb4cde7ca0ba1eb
+type_tags:
+- utility
+toolkit:
+  go:
+    package_name: moneylion.com/security/bitrise-step-aws-secrets-manager
+is_requires_admin_user: true
+is_always_run: false
+is_skippable: false
+inputs:
+- aws_access_key_id: ""
+  opts:
+    description: The AWS Access Key ID. If you are supplying this input, ensure the
+      variable is named exactly `AWS_ACCESS_KEY_ID`.
```

Comment:
NOTE: Bitrise Workflow Editor shows the input's title and summary (description replaces summary if you click on the input) by default. This means by default users will not see that the value of this input needs to be `$AWS_ACCESS_KEY_ID`.

It would be better to set the required env vars based on the provided input values. This way users do not need to know what envs are required, they just configure step inputs.

The same applies to `aws_secret_access_key`.

If these inputs are required please add the `is_required: true property`.

Reply:
Thanks @godrei  for the suggestion. I'm also wondering what's the best way here. If we were to set the required env var, would it be something like:

```yaml
- aws_access_key_id: $AWS_ACCESS_KEY_ID
  opts:
    is_required: true
    is_sensitive: true
```

But this means user would always need to supply `AWS_ACCESS_KEY_ID`, which is not the case, since we want to allow user to use shared configuration via `AWS_PROFILE` as well, without specifying static credentials.

Internally, the Step does not use `aws_access_key_id` nor `aws_secret_access_key` at all. It merely passes the environment variables, which have to be named exactly, to AWS SDK for Go. We kind of rely implicitly on the requirements for AWS SDK for Go here.

In this case, is it possible to set the input as both:
- not required; and
- with a default value for environment variable, i.e. provide a default value for `AWS_ACCESS_KEY_ID`?